HTMLScriptElement.textContent/innerText - docs for TrustedTypes

hamishwillee opened the pull request:
This updates the TrustedType documentation for `HTMLScriptElement.textContent` so that it includes boilerplate and security considerations.

These was inherited from node but is now in IDL - see https://w3c.github.io/trusted-types/dist/spec/#enforcement-in-scripts

Note that this essentially copies [`HTMLScriptElement.text`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLScriptElement/text) which was updated in #41128. The only minor difference is the note about the fact it is also defined on Node and so can be used with other elements.

Project tracking in #37518

hamishwillee committed:
HTMLScriptElement/Node.textContent - docs for TT

hamishwillee committed:
HTMLScriptElement - top level add down links

hamishwillee committed:
HTMLScriptElement.innerText and tweaks

hamishwillee committed:
Apply suggestion from @hamishwillee

hamishwillee commented on files/en-us/web/api/htmlscriptelement/innertext/index.md:
```diff
@@ -0,0 +1,167 @@
+
+The **`innerText`** property of the {{domxref("HTMLScriptElement")}} interface represents the text content inside the {{HTMLElement("script")}} element as though it were rendered text.
+
+When the property is set, the input is normalized the same way as any other element's `innerText` — whitespace is collapsed and `\n` is converted into line breaks.
+This produces the textual representation a user would obtain if a `<script>` element had visible rendering and they were to select its contents and copy them to the clipboard.
```
This does not appear to be true. If you look at my second example below the output for both methods is the same, which would means that the section on `text` vs `textContent` vs `innerText` is also wrong. I will check

fred-wang replied:
@hamishwillee I'm not aware of any difference. Normally the `HTMLScriptElement` specialization is just to do the TT enforcement before. Do you mean enabling TT causes a behavior change here?

hamishwillee replied:
> Do you mean enabling TT causes a behavior change here?

@fred-wang No. Definitely not!

What I'm asking is whether the `innerText` returns exactly the same value as `text` and `textContent`.

In other elements the returned value may be different because the `innerText` is supposed to be like a rendered version of the DOM, with whitespace removed. But for script elements you can't have inner elements. So my suspicion is that _for script elements_ the result is the same as the other properties and always has been.

The test certainly shows that the output seems the same.

I think you are saying is yes, as far as you know.

hamishwillee commented on files/en-us/web/api/htmlscriptelement/index.md:
```diff
@@ -33,22 +36,30 @@ _Inherits properties from its parent, {{domxref("HTMLElement")}}._
+  - : A string representing the URL of an external script; this can be used as an alternative to embedding a script directly within a document.
+    It reflects the `src` attribute of the {{HTMLElement("script")}} element.
 - {{domxref("HTMLScriptElement.text")}}
-  - : A string that joins and returns the contents of all {{domxref("Text")}} nodes inside the {{HTMLElement("script")}} element (ignoring other nodes like comments) in tree order. On setting, it acts the same way as the {{domxref("Node.textContent")}} property.
```
Not really relevant here.

hamishwillee committed:
All the text props same. Make similar

hamishwillee committed:
Apply suggestion from @hamishwillee

hamishwillee commented:
@wbamberg This should be good to look at now.

It turns out that I was wrong about the properties `textContent`, `innerText`, and  `text`, being different - testing shows that setting and getting has the same result in all of them. The spec is also the same for `innerText` and `textContent`. The spec for  `text` is very similar (it lacks steps for getter, which I think is an omission)
They aren't different in testing prior to TT either on Firefox.
I think that this is because it is a script element - the script element isn't rendered and can only contain a text element, so most of the differences in definition simply wouldn't apply even in the old spec wording.

Anyway, because of this I have
- removed the comparison section and added a statement they are the same 
- Security considerations and Examples sections all refer to the version in textContent doc
- Descriptions are almost the same. The difference is, if relevant, a comment about the version in the base class.

hamishwillee committed:
Apply suggestion from @hamishwillee

hamishwillee committed:
Update files/en-us/web/api/htmlscriptelement/innertext/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

hamishwillee committed:
Update files/en-us/web/api/htmlscriptelement/text/index.md

hamishwillee committed:
Update files/en-us/web/api/htmlscriptelement/textcontent/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

hamishwillee committed:
Apply suggestion from @wbamberg

Co-authored-by: wbamberg <will@bootbonnet.ca>

hamishwillee committed:
Fix enforcing trusted type > types typo

hamishwillee committed:
Apply suggestion from @wbamberg

Co-authored-by: wbamberg <will@bootbonnet.ca>

hamishwillee committed:
Fix case everywhere of Trusted Types JavaScript API

wbamberg reviewed (approved):
👍 thanks, and sorry to be slow...
